jax/_src/test_util.py

```diff
@@ -808,23 +816,25 @@ def assertMultiLineStrippedEqual(self, expected, what):
                               msg=f"Found\n{what}\nExpecting\n{expected}")
 
   def _CompileAndCheck(self, fun, args_maker, *, check_dtypes=True,
-                       rtol=None, atol=None, check_cache_misses=True):
+                       rtol=None, atol=None, check_cache_misses=True,
+                       post_process=None):
```

Comment:
IIUC this new argument is only used in one (parameterized) test. It seems a little heavyweight to change this low-level method, depended on by all tests, just to make one new parameterized test work. Plus, adding behavior via kwargs is a smell, as you like to say!

How about we just make that one test in random_test.py call its own version of _CompileAndCheck, rather than editing this one?

Reply:
Thanks calling this out. You knew I'd agree! I fixed this in a way that required neither the kwarg nor a one-off `_CompileAndCheck` variant, thankfully.